src/main/java/com/example/hubtjobsreactive/controller/HiringPostController.java

```diff
@@ -11,30 +15,76 @@
 
     private final HiringPostServiceImpl hiringPostService;
 
+    private final MeterRegistry meterRegistry;
+
     @GetMapping("/posts/{id}")
     public Mono<HiringPostResponse> getPostById(@PathVariable Long id) {
-        return hiringPostService.getHiringPostById(id);
+        AtomicReference<Instant> start = new AtomicReference<>();
+        return hiringPostService.getHiringPostById(id)
+                .doOnSubscribe(p -> {
+                    start.set(Instant.now());
+                    applyMetrics(List.of(Tag.of("method", "GET"), Tag.of("uri", "/posts/" + id)));
+                })
+                .doOnSuccess(p -> {
+                    Duration delta = Duration.between(start.get(), Instant.now());
+                    applyTimerMetrics(List.of(Tag.of("method", "GET"), Tag.of("uri", "/posts/" + id)), delta);
+                });
     }
 
     @GetMapping("/posts")
     public Flux<HiringPostResponse> searchPost(@RequestParam(name = "keyword", required = false) String keyword) {
-        if (keyword == null) {
-            return hiringPostService.getAllPost();
-        }
-        return hiringPostService.searchPost(keyword);
+        AtomicReference<Instant> start = new AtomicReference<>();
+        return hiringPostService.searchPost(keyword)
+                .doOnSubscribe(p -> {
+                    start.set(Instant.now());
+                    applyMetrics(List.of(Tag.of("method", "GET"), Tag.of("uri", "/posts")));
+                })
+                .doOnComplete(() -> {
+                    Duration delta = Duration.between(start.get(), Instant.now());
+                    applyTimerMetrics(List.of(Tag.of("method", "GET"), Tag.of("uri", "/posts")), delta);
+                });
     }
 
     @PostMapping(value = "poster/posts", consumes = MediaType.MULTIPART_FORM_DATA_VALUE,
             produces = MediaType.APPLICATION_STREAM_JSON_VALUE)
     public Mono<HiringPostResponse> createHiringPost(@Valid HiringPostRequest request,
                                                      @RequestPart(value = "image", required = false) FilePart filePart) {
-        return hiringPostService.createHiringPost(request, filePart);
+        AtomicReference<Instant> start = new AtomicReference<>();
+        return hiringPostService.createHiringPost(request, filePart)
+                .doOnSubscribe(p -> {
+                    start.set(Instant.now());
+                    applyMetrics(List.of(Tag.of("method", "POST"), Tag.of("uri", "/poster/posts")));
+                })
+                .doOnSuccess(x -> {
+                    Duration delta = Duration.between(start.get(), Instant.now());
+                    applyTimerMetrics(List.of(Tag.of("method", "POST"), Tag.of("uri", "/poster/posts")), delta);
+                });
+    }
+
+    private void applyMetrics(Iterable<Tag> tags) {
+        Counter.builder("counter-rest-api")
+                .tags(tags)
+                .register(meterRegistry)
+                .increment(1);
     }
+
```

Comment:
u might need a separate interface+implementation for metrics, because it this block of code will be reused on many controllers/services

Reply:
@huuquannguyen remember to extract "counter-rest-api", "timer-rest-api" to method param